tests/cmds/test_devices.py

```diff
@@ -0,0 +1,478 @@
+def archives_list_generator(mocker):
+    yield TEST_ARCHIVES_RESPONSE
+
+
+@pytest.fixture
+def devices_list_generator(mocker):
+    return [TEST_COMPUTER_PAGE]
+
+
+@pytest.fixture
+def users_list_generator(mocker):
+    return [TEST_USERS_LIST_PAGE]
+
+
+@pytest.fixture
+def backupusage_response(mocker):
+    return _create_py42_response(mocker, TEST_BACKUPUSAGE_RESPONSE)
+
+
+@pytest.fixture
+def empty_backupusage_response(mocker):
+    return _create_py42_response(mocker, TEST_EMPTY_BACKUPUSAGE_RESPONSE)
+
+
+@pytest.fixture
+def device_info_success(cli_state, device_info_response):
+    cli_state.sdk.devices.get_by_id.return_value = device_info_response
+
+
+@pytest.fixture
+def archives_list_success(cli_state, archives_list_generator):
+    cli_state.sdk.archive.get_all_by_device_guid.return_value = archives_list_generator
+
+
+@pytest.fixture
+def deactivate_device_success(cli_state, deactivate_response):
+    cli_state.sdk.devices.deactivate.return_value = deactivate_response
+
+
+@pytest.fixture
+def deactivate_device_not_found_failure(cli_state):
+    cli_state.sdk.devices.deactivate.side_effect = Py42NotFoundError(HTTPError())
+
+
+@pytest.fixture
+def deactivate_device_in_legal_hold_failure(cli_state):
+    cli_state.sdk.devices.deactivate.side_effect = Py42BadRequestError(HTTPError())
+
+
+@pytest.fixture
+def deactivate_device_not_allowed_failure(cli_state):
+    cli_state.sdk.devices.deactivate.side_effect = Py42ForbiddenError(HTTPError())
+
+
+@pytest.fixture
+def backupusage_success(cli_state, backupusage_response):
+    cli_state.sdk.devices.get_by_id.return_value = backupusage_response
+
+
+@pytest.fixture
+def empty_backupusage_success(cli_state, empty_backupusage_response):
+    cli_state.sdk.devices.get_by_id.return_value = empty_backupusage_response
+
+
+@pytest.fixture
+def get_all_devices_success(cli_state, devices_list_generator):
+    cli_state.sdk.devices.get_all.return_value = devices_list_generator
+
+
+@pytest.fixture
+def get_all_users_success(cli_state, users_list_generator):
+    cli_state.sdk.users.get_all.return_value = users_list_generator
+
+
+def test_deactivate_deactivates_device(runner, cli_state, deactivate_device_success):
+    runner.invoke(
+        cli, ["devices", "deactivate", "--device-id", TEST_DEVICE_ID], obj=cli_state
+    )
+    cli_state.sdk.devices.deactivate.assert_called_once_with(TEST_DEVICE_ID)
+
+
+def test_deactivate_updates_purge_date(
+    runner,
+    cli_state,
+    deactivate_device_success,
+    device_info_success,
+    archives_list_success,
+):
+    runner.invoke(
+        cli,
+        [
+            "devices",
+            "deactivate",
+            "--device-id",
+            TEST_DEVICE_ID,
+            "--purge-date",
+            TEST_PURGE_DATE,
+        ],
+        obj=cli_state,
+    )
+    cli_state.sdk.archive.update_cold_storage_purge_date.assert_called_once_with(
+        TEST_ARCHIVE_GUID, TEST_PURGE_DATE
+    )
+
+
+def test_deactivate_changes_device_name(
```

Comment:
There should be a test that also makes sure that it _doesn't_ change the name when the name is already prefixed with `deactivated_` because I noticed that conditional in the logic. (sorry if there is one and I just am not seeing it)